### PR TITLE
reference: update tutorials about dm wildcard support

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

DM need to update its newly wildcard usage.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v4.0 changes for now)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/2383
- Other reference link(s):

### CODE REVIEW

File: reference/tools/data-migration/features/overview.md [hunk 1]
```diff
@@ -20,15 +27,26 @@ The table routing feature enables DM to replicate a certain table of the upstrea
 ### Parameter configuration
 
 ```yaml
-routes:
+black-white-list:
```
Comment (yikeke) [inline reference/tools/data-migration/features/overview.md]: Warning: You should modify the "Parameter configuration" under the "Black & white table lists" section, not "Table routing" section. See https://github.com/pingcap/docs-cn/pull/2383/files. 

Please update accordingly. @TomShawn

Comment (TomShawn) [reply reference/tools/data-migration/features/overview.md]: Updated, PTAL.

File: reference/tools/data-migration/features/overview.md [hunk 2]
```diff
@@ -102,7 +109,15 @@ The black and white lists filtering rule of the upstream database instance table
 ```yaml
 black-white-list:
   rule-1:
-    do-dbs: ["~^test.*"]         # Starting with "~" indicates it is a regular expression.
+    do-dbs: ["test*"]         # Starting with characters other "~" indicates it is a wildcard;
```
Comment (yikeke) [inline reference/tools/data-migration/features/overview.md]:
```suggestion
    do-dbs: ["test*"]         # Starting with characters other than "~" indicates that it is a wildcard;
```

File: reference/tools/data-migration/features/overview.md [hunk 3]
```diff
@@ -102,7 +109,15 @@ The black and white lists filtering rule of the upstream database instance table
...
+                              # v1.0.4 or later versions support the regular expression rules.
+​    do-tables:
+    - db-name: "test[123]"    # Matches test1, test2, and test3.
+      tbl-name: "t[1-5]"      # Matches t1, t2, t3, t4, and t5.
+    - db-name: "test"
+      tbl-name: "t"
+  rule-2:
+    do-dbs: ["~^test.*"]      # Starting with "~" indicates it is a regular expression.
```
Comment (yikeke) [inline reference/tools/data-migration/features/overview.md]:
```suggestion
    do-dbs: ["~^test.*"]      # Starting with "~" indicates that it is a regular expression.
```

**Pull request merged**